distribution/tools/entitlement-agent/impl/src/main/java/org/elasticsearch/entitlement/instrumentation/impl/InstrumenterImpl.java

```diff
@@ -147,9 +145,9 @@ public void visitCode() {
         private void pushEntitlementChecksObject() {
             mv.visitMethodInsn(
                 INVOKESTATIC,
-                Type.getDescriptor(EntitlementProvider.class),
+                "org/elasticsearch/entitlement/api/EntitlementProvider",
                 "checks",
-                Type.getMethodDescriptor(Type.getType(EntitlementChecks.class)),
+                "()Lorg/elasticsearch/entitlement/api/EntitlementChecks;",
```

Comment:
@jdconrad - I want to draw your attention to these changes, since I don't think you like this style of code very much.

Reply:
This change (kind of) eliminated the agent's compile-time dependency on the bridge. The agent still actually depends on the bridge to have a certain class with a certain method, so the dependency is still there, but now it's a runtime dependency. Is that better? 🤷

Reply:
I suppose if we prefer to make this a compile-time dependency, it's simple enough to change in a subsequent PR.